src/plonk/mod.rs

```diff
@@ -485,39 +487,56 @@ impl<F: PrimeField> TableData<F> {
                 .chain(selector_columns)
                 .collect::<Vec<_>>()[..],
         );
-        let num_selector = self.cs.num_selectors();
-        let num_fixed = self.cs.num_fixed_columns();
-        let num_advice = self.cs.num_advice_columns();
 
-        // we use the same challenge y for combining custom gates and multiple lookup arguments
-        // need first commit all witness before generating y
-        let y = Expression::Polynomial(Query {
-            index: num_selector + num_fixed + num_advice,
-            rotation: Rotation(0),
-        });
         // suppose we have n polynomial expression: p_1,p_2,...,p_n
         // we combined them together as one: combined_poly = p_1*y^{n-1}+p_2*y^{n-2}+...+p_n
-        let combined_poly: MultiPolynomial<C::ScalarExt> = self
+        let num_gates = self
             .cs
             .gates()
             .iter()
             .flat_map(|gate| gate.polynomials().iter())
-            .map(|expr| Expression::from_halo2_expr(expr, num_selector, num_fixed))
-            .fold(Expression::Constant(F::ZERO), |acc, expr| {
-                Expression::Sum(
-                    Box::new(expr),
-                    Box::new(Expression::Product(Box::new(acc), Box::new(y.clone()))),
-                )
-            })
-            .expand();
+            .collect::<Vec<_>>()
+            .len();
```

Comment:
Don't need collect and allocate `Vec` for this, just use [count](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.count)